packages/react-reconciler/src/__tests__/useSyncExternalStore-test.js

```diff
@@ -292,4 +353,88 @@ describe('useSyncExternalStore', () => {
       );
     },
   );
+
+  it('regression: doesnt loop for only changing store reference', async () => {
+    let store = {validationResults: new Map(), version: 0};
+    let listeners = [];
+
+    const ExternalStore = {
+      set(value) {
+        // Change the store ref, but not the value.
+        store = {...store};
+        emitChange();
+      },
+      subscribe(listener) {
+        listeners = [...listeners, listener];
+        return () => {
+          listeners = listeners.filter(l => l !== listener);
+        };
+      },
+      getSnapshot() {
+        return store;
+      },
+    };
+
+    function emitChange() {
+      for (const listener of listeners) {
+        listener();
+      }
+    }
+
+    function useBug<TValue>(path: string): SRConfigPathState<TValue> {
+      const {value} = useSyncExternalStore(
```

Comment:
store doesn't have a `value` field (?)

Reply:
yeah, need to rename `validationResults` to `value`